File: ShopAppSpring/src/main/java/com/duchung/shopappspring/dtos/OrderDTO.java
package com.duchung.shopappspring.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotEmpty;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class OrderDTO {

    @JsonProperty("user_id")
    @Min(value = 0, message = "User id must be greater than 0")
    private Long userID;

    @JsonProperty("fullname")
    private String fullName;

    private String email;

    @JsonProperty("phone_number")
    private String phoneNumber;

    private String address;

    private String note;

    @JsonProperty("total_money")
    @Min(value = 0, message = "Total money must be greater than 0")
    private double totalMoney;

    @JsonProperty("shipping_method")
    @NotEmpty(message = "Shipping method can't be blank")
    private String shippingMethod;

    @JsonProperty("shipping_address")
    private String shippingAddress;

    @JsonProperty("payment_method")
    @NotEmpty(message = "Payment method can't be blank")
    private String paymentMethod;

    @JsonProperty("cart_items")
    private List<CartItemDTO> cartItems;
}

File: shopapp-angular/src/app/dtos/order.dto.ts
export class OrderDTO {
  user_id: number;
  fullname: string;
  email: string;
  phone_number: string;
  address: string;
  note: string;
  total_money: number;
  shipping_method: string;
  shipping_address: string;
  payment_method: string;
  cart_items: { product_id: number; quantity: number }[];

  constructor(data: any) {
    this.user_id = data.user_id;
    this.fullname = data.fullname;
    this.email = data.email;
    this.phone_number = data.phone_number;
    this.address = data.address;
    this.note = data.note;
    this.total_money = data.total_money;
    this.shipping_address = data.shipping_address;
    this.shipping_method = data.shipping_method;
    this.payment_method = data.payment_method;
    this.cart_items = data.cart_items;
  }
}
